experiments/evals/run_base_model_evals.py

```diff
@@ -0,0 +1,125 @@
+"""
+Base model evaluations across multiple LLMs.
+
+This experiment evaluates OLMO Base 8B, LLAMA 3.1 8B, Deeper Starling 8B,
+MAP-NEO 7B, and Amber Base 7B models on CORE_TASKS (augmented with GPQA) as well
+as dedicated MMLU 0-shot and 5-shot configurations.
+"""
+
+from experiments.evals.evals import default_eval
+from experiments.evals.resource_configs import SINGLE_TPU_V6E_8
+from experiments.evals.task_configs import CORE_TASKS
+from experiments.models import amber_base_7b, llama_3_1_8b, map_neo_7b, olmo_2_base_8b
+from marin.evaluation.evaluation_config import EvalTaskConfig
+from marin.execution.executor import executor_main
+
+# Model path for deeper starling
+deeper_starling_path = "gs://marin-us-central2/checkpoints/tootsie-8b-deeper-starling/hf/step-1419999"
+
+# Add GPQA to CORE_TASKS
+core_tasks_with_gpqa = (
+    EvalTaskConfig(
+        "leaderboard_gpqa",
+        0,
+        task_alias="gpqa_0shot",
+    ),
+    *CORE_TASKS,
+)
+
+# Create reusable EvalTaskConfig objects
+mmlu_0shot_config = [EvalTaskConfig("mmlu", 0, task_alias="mmlu_0shot")]
+mmlu_5shot_config = [EvalTaskConfig("mmlu", 5, task_alias="mmlu_5shot")]
+
+# Set up evaluations for core tasks (including GPQA)
+olmo_base_core = default_eval(step=olmo_2_base_8b, resource_config=SINGLE_TPU_V6E_8, evals=core_tasks_with_gpqa)
+
+llama_core = default_eval(step=llama_3_1_8b, resource_config=SINGLE_TPU_V6E_8, evals=core_tasks_with_gpqa)
+
+deeper_starling_core = default_eval(
+    step=deeper_starling_path,
+    resource_config=SINGLE_TPU_V6E_8,
+    evals=core_tasks_with_gpqa,
+)
+
+map_neo_core = default_eval(
+    step=map_neo_7b,
+    resource_config=SINGLE_TPU_V6E_8,
+    evals=core_tasks_with_gpqa,
+)
+
+amber_base_core = default_eval(
+    step=amber_base_7b,
+    resource_config=SINGLE_TPU_V6E_8,
+    evals=core_tasks_with_gpqa,
+)
+
+# Set up evaluations for MMLU 0-shot (run separately for macro avg calculation)
+olmo_base_mmlu_0shot = default_eval(step=olmo_2_base_8b, resource_config=SINGLE_TPU_V6E_8, evals=mmlu_0shot_config)
```

Comment:
Can we package the mmlu evals into a function like `default_eval`?  It seems like we're going to want a single function that just does all the evals we care about (maybe with flags) for the large number of models we're going to see next...and we want O(# models) work, not O(# models x # evals) work.

Reply:
Done